picohttpparser.c

```diff
@@ -359,9 +359,9 @@ static const char *parse_request(const char *buf, const char *buf_end, const cha
 
     /* parse request line */
     ADVANCE_TOKEN(*method, *method_len);
-    ++buf;
+    while (*buf == ' ') ++buf;
```

Comment:
At the end of ADVANCE_TOKEN, we know that `*buf` is a whitespace. Therefore, may something like below?

```c
do {
    ++buf;
} (while *buf == ' ');
```